DSCResources/MSFT_xIisLogging/MSFT_xIisLogging.psm1

```diff
@@ -0,0 +1,290 @@
+#requires -Version 4.0 -Modules CimCmdlets
+
+# Load the Helper Module
+Import-Module -Name "$PSScriptRoot\..\Helper.psm1"
+
+# Localized messages
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData -StringData @'
+VerboseSetTargetUpdateLogPath = LogPath is not in the desired state and will be updated.
+VerboseSetTargetUpdateLogFlags = LogFlags do not match and will be updated.
+VerboseSetTargetUpdateLogPeriod = LogPeriod is not in the desired state and will be updated.
+VerboseSetTargetUpdateLogTruncateSize = TruncateSize is not in the desired state and will be updated.
+VerboseSetTargetUpdateLoglocalTimeRollover = LoglocalTimeRollover is not in the desired state and will be updated.
+VerboseSetTargetUpdateLogFormat = LogFormat is not in the desired state and will be updated
+VerboseTestTargetFalseLogPath = LogPath does match desired state.
+VerboseTestTargetFalseLogFlags = LogFlags does not match desired state.
+VerboseTestTargetFalseLogPeriod = LogPeriod does not match desired state.
+VerboseTestTargetFalseLogTruncateSize = LogTruncateSize does not match desired state.
+VerboseTestTargetFalseLoglocalTimeRollover = LoglocalTimeRollover does not match desired state.
+VerboseTestTargetFalseLogFormat = LogFormat does not match desired state.
+WarningLogPeriod = LogTruncateSize has is an input as will overwrite this desired state.
+WarningIncorrectLogFormat = LogFormat is not W3C, as a result LogFlags will not be used. 
+'@
+}
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $LogPath,
+                
+        [String[]]
+        [ValidateSet('Date','Time','ClientIP','UserName','SiteName','ComputerName','ServerIP','Method','UriStem','UriQuery','HttpStatus','Win32Status','BytesSent','BytesRecv','TimeTaken','ServerPort','UserAgent','Cookie','Referer','ProtocolVersion','Host','HttpSubStatus')]
```

Comment:
This isn't strictly required but it is conventional to put the parameter type declaration ([String] etc) after any [Validate*] declarations. Are we able to reverse the order?